core/iwasm/libraries/lib-wasi-threads/lib_wasi_threads_wrapper.c

```diff
@@ -136,6 +177,7 @@ thread_spawn_wrapper(wasm_exec_env_t exec_env, uint32 start_arg)
     deallocate_thread_id(thread_id);
 
 thread_preparation_fail:
+    wasm_runtime_free(*new_c_api_func_imports);
```

Comment:
If I understand correctly this is already implied by the next two lines - as deinstantiate frees c_api_func_imports in line 2265.